packages/modal/src/useModal.ts

```diff
@@ -48,37 +48,33 @@ export function useModal(
   const [idPrefix] = useState(_id || seed(`modal_${PACKAGE_VERSION}`));
   const titleId = `${idPrefix}--title`;
   const contentId = `${idPrefix}--content`;
-  const isModalMousedDownRef = useRef(false);
 
   const closeModal = (event: KeyboardEvent | MouseEvent) => {
     onClose && onClose(event);
   };
 
-  const getBackdropProps = ({ onMouseUp, ...other } = {} as any) => {
+  const getBackdropProps = ({ onClick, ...other } = {} as any) => {
     return {
-      onMouseUp: composeEventHandlers(onMouseUp, (event: MouseEvent) => {
-        if (!isModalMousedDownRef.current) {
+      onClick: composeEventHandlers(onClick, (event: MouseEvent) => {
+        const target = event.target as Element;
+
+        if (target?.getAttribute('data-garden-container-id') === 'containers.modal') {
```

Comment:
Rather than looking at the `data-garden-container-id` attribute, could we compare target to the value stored in `modalRef.current`?

Reply:
Probably, feel free to to take over the PR.

Reply:
I agree that would be nicer.

Reply:
Looks like the `modalRef.current` references the focus jail element and not the modal backdrop. I've kept @sunesimonsen's original approach regarding checking the data attribute. I believe this is the ideal approach.

Reply:
I was misunderstanding this logic. This is looking for the backdrop _not_ the modal wrapper. This logic is fine for me.